Assets/Vimeo/Scripts/Services/VimeoUploader.cs

```diff
@@ -0,0 +1,176 @@
+using System.IO;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+using SimpleJSON;
+using Vimeo;
+
+namespace Vimeo
+{
+    class VimeoUploader : MonoBehaviour
+    {
+        //Events
+        public delegate void UploadAction(string status, float progress = 0.0f);
+        public event UploadAction OnUploadProgress;
+        public event UploadAction OnUploadError;
+        public event UploadAction OnUploadComplete;
+
+        public delegate void UploadEvent(VideoChunk chunk, string msg = "");
+        public event UploadEvent OnChunckUploadComplete;
+        public event UploadEvent OnChunckUploadError;
+
+        //Private members
+        private Queue<VideoChunk> myChunks;
+        public VimeoApi vimeoApi;
+        //TODO: In the future this will be stored in a hash table to provide batch uploading
+        private string file;
+        private string vimeo_url;
+        private FileInfo fileInfo;
+
+
+        //Public members
```

Comment:
dont think you need to comment this area out. save comments to explain things that aren't innately clear already.

Reply:
Yeah, sorry I cleaned up all the comments/whitespaces in the last commit